### PR TITLE
Fixed Issue 2743 of notifications in peer activity

### LINKED ISSUE #2743: Review: Notifications seem to be about picture I have not reviewed yet
I just tried the review feature for the first time today. I reviewed the picture of a 2016 senator, all perfect, great picture with relevant categories, so I tapped "seems fine" at each step.

After sending the thank, two notification appeared. Problem: They seem to be about the next picture (Pedro Americo drawing).

Other apparent problem: "Requesting category check" appears despite me tapping to say that the categories are fine. (my [contributions](https://commons.wikimedia.org/wiki/Special:Contributions/Syced) do not show any activity for that period, though)
Screenshot_20190325-145038

### PR DESCRIPTION
**Pull request by vanshikaarora**
**Description (required)**
The issue was due to same publish progress function being called for Reporting Wrong Category and for  Sending Thanks hence the wrong progress (report wrong category) was shown instead of sending thank in notification.
>After sending the thank, two notification appeared. Problem: They seem to be about the next picture

This was due to updation of filename after the last question. 

Fixes #2743 Notifications seem to be about picture I have not reviewed yet

### Changes Made
- Added separate publish progress functions for Reporting Wrong Category and for  Sending Thanks
- Send file name while calling these functions

### CODE REVIEW

File: app/src/main/java/fr/free/nrw/commons/review/ReviewController.java [hunk 1]
```diff
@@ -151,8 +152,24 @@ private void publishProgress(@NonNull Context context, int i) {
...
+                        .bigText(message))
+                .setSmallIcon(R.drawable.ic_launcher)
+                .setProgress(messages.length, i, false)
+                .setOngoing(true);
+        notificationManager.notify(NOTIFICATION_SEND_THANK, notificationBuilder.build());
+    }
+
     @SuppressLint("CheckResult")
-    public void sendThanks(@NonNull Activity activity) {
+    public void sendThanks(@NonNull Activity activity, String fileName) {
```
Comment (nicolas-raoul) [inline app/src/main/java/fr/free/nrw/commons/review/ReviewController.java]: Sorry I had not seen this earlier: Thanks->Thank

**Comment by vanshikaarora** [comment]
@nicolas-raoul I have made the changes. Kindly review :)

**Comment by nicolas-raoul** [comment]
Don't forget the first rule of https://github.com/commons-app/apps-android-commons/wiki/Code-style :-)
Even if paraphrasing, it is actually helpful to readers. Source code written once is read by hundreds of people, so it is worth the effort.

File: app/src/main/java/fr/free/nrw/commons/review/ReviewController.java [hunk 2]
```diff
@@ -151,8 +152,30 @@ private void publishProgress(@NonNull Context context, int i) {
         notificationManager.notify(NOTIFICATION_CHECK_CATEGORY, notificationBuilder.build());
     }
 
+    private void publishProgressForSendingThank(Context context, int i){
```
Comment (neslihanturan) [inline app/src/main/java/fr/free/nrw/commons/review/ReviewController.java]: Please add Javadoc for methods you added.

File: app/src/main/java/fr/free/nrw/commons/review/ReviewController.java [hunk 3]
```diff
@@ -151,8 +152,30 @@ private void publishProgress(@NonNull Context context, int i) {
...
+        int[] messages = new int[]{R.string.getting_edit_token, R.string.send_thank_send};
```
Comment (neslihanturan) [inline app/src/main/java/fr/free/nrw/commons/review/ReviewController.java]: idOfMessages or similar namincg would be better for messages variable.

**Comment by vanshikaarora** [comment]
@neslihanturan I have made the changes. Kindly review :)

File: app/src/main/java/fr/free/nrw/commons/review/ReviewController.java [hunk 4]
```diff
@@ -135,24 +141,56 @@ public void reportWrongCategory(@NonNull Activity activity) {
...
 
-    private void publishProgress(@NonNull Context context, int i) {
-        int[] messages = new int[]{R.string.getting_edit_token, R.string.check_category_adding_template};
+    /**
+     * @param context
+     * @param i progress as an integer
+    * While reportWrongCategory() is in background notify the user about current progress in the newtwork request*/
+
+    private void publishProgressForWrongCategory(@NonNull Context context, int i) {
+        int[] listOfMessages = new int[]{R.string.getting_edit_token, R.string.check_category_adding_template};
```
Comment (neslihanturan) [inline app/src/main/java/fr/free/nrw/commons/review/ReviewController.java]: What I meant was, this is not listOfMessages. This is list of integers which are ids of messages.

Comment (vanshikaarora) [reply app/src/main/java/fr/free/nrw/commons/review/ReviewController.java]: I am sorry, I have misunderstood that before :(. I have made the changes in the latest commit.

**Comment by neslihanturan** [comment]
Closing this issue since there are no changes at all, please feel free to reopen it with differences @vanshikaarora , thanks:)

**Pull request closed without merging**